cli/pbmctl/main_test.go

```diff
@@ -61,72 +62,80 @@ func TestListAgents(t *testing.T) {
 
 	sort.Slice(clients, func(i, j int) bool { return clients[i].Id < clients[j].Id })
 	want := []*api.Client{
-		&api.Client{
-			Version:        0,
-			Id:             "127.0.0.1:17000",
-			NodeType:       "MONGOS",
-			NodeName:       "127.0.0.1:17000",
-			ReplicasetName: "",
-		},
 		&api.Client{
 			Version:        0,
 			Id:             "127.0.0.1:17001",
-			NodeType:       "MONGOD_SHARDSVR",
+			NodeType:       "NODE_TYPE_MONGOD_SHARDSVR",
 			ReplicasetName: "rs1",
 		},
 		&api.Client{
 			Version:        0,
 			Id:             "127.0.0.1:17002",
-			NodeType:       "MONGOD_SHARDSVR",
+			NodeType:       "NODE_TYPE_MONGOD_SHARDSVR",
 			ReplicasetName: "rs1",
 		},
 		&api.Client{
 			Version:        0,
 			Id:             "127.0.0.1:17003",
-			NodeType:       "MONGOD_SHARDSVR",
+			NodeType:       "NODE_TYPE_MONGOD_SHARDSVR",
 			ReplicasetName: "rs1",
 		},
 		&api.Client{
 			Version:        0,
 			Id:             "127.0.0.1:17004",
-			NodeType:       "MONGOD_SHARDSVR",
+			NodeType:       "NODE_TYPE_MONGOD_SHARDSVR",
 			ReplicasetName: "rs2",
 		},
 		&api.Client{
 			Version:        0,
 			Id:             "127.0.0.1:17005",
-			NodeType:       "MONGOD_SHARDSVR",
+			NodeType:       "NODE_TYPE_MONGOD_SHARDSVR",
 			ReplicasetName: "rs2",
 		},
 		&api.Client{
 			Version:        0,
 			Id:             "127.0.0.1:17006",
-			NodeType:       "MONGOD_SHARDSVR",
+			NodeType:       "NODE_TYPE_MONGOD_SHARDSVR",
 			ReplicasetName: "rs2",
 		},
 		&api.Client{
 			Version:        0,
 			Id:             "127.0.0.1:17007",
-			NodeType:       "MONGOD_CONFIGSVR",
+			NodeType:       "NODE_TYPE_MONGOD_CONFIGSVR",
 			NodeName:       "127.0.0.1:17007",
 			ReplicasetName: "csReplSet",
 		},
+		&api.Client{
+			Version:        0,
+			Id:             "127.0.0.1:17000",
+			NodeType:       "NODE_TYPE_MONGOS",
+			NodeName:       "127.0.0.1:17000",
+			ReplicasetName: "",
+		},
 	}
 
 	for i, client := range clients {
 		if client.Version != want[i].Version {
 			t.Errorf("Invalid client version. Got %v, want %v", client.Version, want[i].Version)
 		}
-		if client.Id != want[i].Id {
+		if client.NodeType != "NODE_TYPE_MONGOS" && client.Id != want[i].Id {
 			t.Errorf("Invalid client id. Got %v, want %v", client.Id, want[i].Id)
+		} else if client.NodeType == "NODE_TYPE_MONGOS" {
+			// fix for mongos having hostname:port instead of ip:port
+			hostname, _ := os.Hostname()
+			wantHostPort := strings.SplitN(want[i].Id, ":", 2)
+			wantHostPortId := hostname + ":" + wantHostPort[1]
+			if client.Id != wantHostPortId && client.Id != want[i].Id {
```

Comment:
@percona-csalguero: I wanted to check this fix is valid or if the Id is wrong.

Mongos is returning "<hostname>:10000" on my test environment when the test expects "<ip>:10000".